src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -431,10 +498,34 @@ protected Query doToQuery(QueryShardContext context) {
             fieldDimension = modelMetadata.getDimension();
             knnEngine = modelMetadata.getKnnEngine();
             spaceType = modelMetadata.getSpaceType();
+            if (modelMetadata.getMethodComponentContext() != null) {
```

Comment:
Need inputs on whether this check is needed

Reply:
We added the method component context to `ModelMetadata` in 2.13.  Therefore, there could be a situation where a model is created before 2.13, the cluster is updated to 2.13 or later, and then that method will return null.